src/dataset/mod.rs

```diff
@@ -134,3 +134,62 @@ mod tests {
         assert_eq!(*m[1][3], 9);
     }
 }
+
+impl<X: Copy + std::fmt::Debug, Y: Copy + std::fmt::Debug> std::fmt::Display for Dataset<X, Y> {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        struct Target<Y> {
+            name: String,
+            value: Y,
+        }
+        struct Feature<X> {
+            name: String,
+            value: X,
+        }
+        struct DataPoint<X, Y> {
+            labels: Vec<Target<Y>>,
+            features: Vec<Feature<X>>,
+        }
+        impl<X: Copy + std::fmt::Debug, Y: Copy + std::fmt::Debug> std::fmt::Display for DataPoint<X, Y> {
+            fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
```

Comment:
Can you add a simple unit test that covers this function? For example, I am getting an error when I try to print Iris Dataset:
```
  #[test]
  fn display() {
      let dataset = iris::load_dataset();

      println!("{}", dataset);
  }
````

Reply:
Hm... I see the problem. I assumed (favourite word right there) that you would have a vector of targets per sample, and not use different integer values to represent different targets.
So, for a single sample example:
feature_vec * weight_vec = target_vec = [0, 1, 0, 0]
not:
feature_vec * weight_vec = target_int = 2

This worked in my case since I only had one target

Reply:
Yes, `data` is a flattened matrix of size N x P, where N is number of samples and P is number of features per sample, `target` is a 1d array with target values of size P. I think printing raw values of `target` is a good solution.